packages/fxa-settings/src/models/hooks.ts

```diff
@@ -95,6 +97,40 @@ export function useRelier() {
   return relierFactory.getRelier();
 }
 
+export function useUrlSearchContext() {
+  let { urlSearchContext } = useContext(AppContext);
+  if (urlSearchContext == null) {
+    throw new Error('urlSearchContext never initialized!')
+  }
+  return urlSearchContext;
+}
+
+export function useLocationStateContext() {
+  const location = useLocation();
+  if (location == null) {
+    throw new Error('Location missing!')
+  }
+
+  return new GenericContext((location.state || {}) as Record<string, unknown>);
+}
+
+
+export function useNotifier() {
+  return {
+    onAccountSignIn(_account:unknown) {
+      // FOLLOW-UP: Not yet implemented.
+    }
+  }
+}
+
+export function useBroker() {
+  return {
+    async invokeBrokerMethod(_name:string, _account:unknown) {
```

Comment:
I forgot where we landed on this, will the broker be performing screen navigations? I can see if we pass in the relier/notifier or get from hook, we would have all the information needed to perform screen transititions.

Reply:
For the navigation, I decided to navigate to the next using the router's navigate method. I mostly did this because the broker is not implemented yet, and I was trying to limit the scope of the PR. I feel like when/if we port the broker, this can be revisited? On a side note, I don't know if there's other missing 'messages' that should be passed at this point that we are overlooking.